commit 0a7640c593a54a6a3f558583d82f8b27c7a7d1d1
Author: Cole Robinson <crobinso@redhat.com>
Date:   Mon Sep 26 11:05:55 2011 -0400

    manager: Properly show vm desc in tooltip w/ special xml characters

diff --git a/src/virtManager/manager.py b/src/virtManager/manager.py
index d1b0d9e..5e9491f 100644
--- a/src/virtManager/manager.py
+++ b/src/virtManager/manager.py
@@ -726,13 +726,17 @@ class vmmManager(vmmGObjectUI):
 
     def _build_vm_row(self, vm):
         row = []
+        desc = None
+        if vm.get_description() is not None:
+            desc = util.xml_escape(vm.get_description())
+
         row.insert(ROW_HANDLE, vm)
         row.insert(ROW_NAME, vm.get_name())
         row.insert(ROW_MARKUP, "")
         row.insert(ROW_STATUS, vm.run_status())
         row.insert(ROW_STATUS_ICON, vm.run_status_icon_name())
         row.insert(ROW_KEY, vm.get_uuid())
-        row.insert(ROW_HINT, vm.get_description())
+        row.insert(ROW_HINT, desc)
         row.insert(ROW_IS_CONN, False)
         row.insert(ROW_IS_CONN_CONNECTED, True)
         row.insert(ROW_IS_VM, True)
@@ -868,6 +872,10 @@ class vmmManager(vmmGObjectUI):
         if self.vm_row_key(vm) not in self.rows:
             return
 
+        desc = None
+        if vm.get_description() is not None:
+            desc = util.xml_escape(vm.get_description())
+
         row = self.rows[self.vm_row_key(vm)]
         row[ROW_NAME] = vm.get_name()
         row[ROW_STATUS] = vm.run_status()
@@ -876,7 +884,7 @@ class vmmManager(vmmGObjectUI):
         row[ROW_MARKUP] = self._build_vm_markup(row)
 
         if config_changed:
-            row[ROW_HINT] = vm.get_description()
+            row[ROW_HINT] = desc
 
         model.row_changed(row.path, row.iter)
 
commit 79447707ace9dc75ad1ef2f86ce1106acccae1f5
Author: Cole Robinson <crobinso@redhat.com>
Date:   Tue Jan 17 15:31:20 2012 -0500

    manager: Again, don't fetch XML on every tick for <description>
    
    The xml_escaping change reverted this.

diff --git a/src/virtManager/manager.py b/src/virtManager/manager.py
index c866236..d1e2d93 100644
--- a/src/virtManager/manager.py
+++ b/src/virtManager/manager.py
@@ -730,9 +730,6 @@ class vmmManager(vmmGObjectUI):
 
     def _build_vm_row(self, vm):
         row = []
-        desc = None
-        if vm.get_description() is not None:
-            desc = util.xml_escape(vm.get_description())
 
         row.insert(ROW_HANDLE, vm)
         row.insert(ROW_NAME, vm.get_name())
@@ -740,7 +737,7 @@ class vmmManager(vmmGObjectUI):
         row.insert(ROW_STATUS, vm.run_status())
         row.insert(ROW_STATUS_ICON, vm.run_status_icon_name())
         row.insert(ROW_KEY, vm.get_uuid())
-        row.insert(ROW_HINT, desc)
+        row.insert(ROW_HINT, util.xml_escape(vm.get_description()))
         row.insert(ROW_IS_CONN, False)
         row.insert(ROW_IS_CONN_CONNECTED, True)
         row.insert(ROW_IS_VM, True)
@@ -876,10 +873,6 @@ class vmmManager(vmmGObjectUI):
         if self.vm_row_key(vm) not in self.rows:
             return
 
-        desc = None
-        if vm.get_description() is not None:
-            desc = util.xml_escape(vm.get_description())
-
         row = self.rows[self.vm_row_key(vm)]
         row[ROW_NAME] = vm.get_name()
         row[ROW_STATUS] = vm.run_status()
@@ -888,7 +881,7 @@ class vmmManager(vmmGObjectUI):
         row[ROW_MARKUP] = self._build_vm_markup(row)
 
         if config_changed:
-            row[ROW_HINT] = desc
+            row[ROW_HINT] = util.xml_escape(vm.get_description())
 
         model.row_changed(row.path, row.iter)
 
